File: algorithms/1_easy/M/03033_modify_the_matrix/solution.cpp
class Solution {
public:
    static vector<vector<int>> modifiedMatrix(const vector<vector<int>>& matrix) {
        const size_t rows{ matrix.size() };
        const size_t cols{ matrix[0].size() };
        auto result{ matrix };

        for (size_t j{ 0 }; j < cols; ++j) {
            int max{ numeric_limits<int>::min() };

            for (size_t i{ 0 }; i < rows; ++i) {
                max = std::max(max, matrix[i][j]);
            }

            for (size_t i{ 0 }; i < rows; ++i) {
                if (matrix[i][j] == -1) {
                    result[i][j] = max;
                }
            }
        }

        return result;
    }
};

File: algorithms/1_easy/M/03033_modify_the_matrix/solution.rs
impl Solution {
    pub fn modified_matrix(mut matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let rows = matrix.len();
        let cols = matrix[0].len();

        for j in 0..cols {
            let mut max = i32::MIN;

            for i in 0..rows {
                max = max.max(matrix[i][j]);
            }

            for i in 0..rows {
                if matrix[i][j] == -1 {
                    matrix[i][j] = max;
                }
            }
        }

        matrix
    }
}
